packages/flutter_tools/lib/src/android/android_device.dart

```diff
@@ -564,6 +564,8 @@ class AndroidDevice extends Device {
         ...<String>['--ez', 'trace-startup', 'true'],
       if (route != null)
         ...<String>['--es', 'route', route],
+      if (debuggingOptions.androidUser.isNotEmpty)
+        ...<String>['--user', debuggingOptions.androidUser],
```

Comment:
When writing your test, note that on line 603 of this file, we pass this `cmd` list to `runAdbCheckedAsync`. Through several layers of indirection, that ends up calling the global `ProcessManager` (which you will have to override in your test with a fake one), passing this command, prepended with some additional adb args, you can see that code in the `adbCommandForDevice` method.

See [FakeProcessManager.list()](https://github.com/flutter/flutter/blob/c3ec1caac2dad49f0c48f6bdbcecd6bf9cab4feb/packages/flutter_tools/test/src/fake_process_manager.dart#L170), which will let you create a fake processmanager that you can set in the test overrides, that will expect a certain sequence of commands. This will be your test assertions, if your test calls the fake process managers with the exact command(s) you set it up with, the test passes; else, your test fails.